orchagent/vnetorch.cpp

```diff
@@ -1967,6 +2228,227 @@ void VNetRouteOrch::updateVnetTunnel(const BfdUpdate& update)
+        return;
+
+    }
+    auto *vrf_obj = vnet_orch_->getTypePtr<VNetVrfObject>(vnet);
+    set<sai_object_id_t> vr_set;
+
+    auto l_fn = [&] (const string& vnet) {
+        auto *vnet_obj = vnet_orch_->getTypePtr<VNetVrfObject>(vnet);
+        sai_object_id_t vr_id = vnet_obj->getVRidIngress();
+        vr_set.insert(vr_id);
+    };
+
+    l_fn(vnet);
+
+    auto primary = route->second.primary;
+    auto secondary = route->second.secondary;
+    auto active_nhg = route->second.nhg_key;
+    NextHopGroupKey nhg_custom("", true);
+    sai_ip_prefix_t pfx;
+    copy(pfx, prefix);
+    NextHopGroupKey nhg_custom_primary = getActiveNHSet( vnet, primary, prefix);
+    NextHopGroupKey nhg_custom_secondary = getActiveNHSet( vnet, secondary, prefix);
+    if (nhg_custom_primary.getSize() > 0)
+    {
+        if (nhg_custom_primary != active_nhg )
+        {
+            if (!hasNextHopGroup(vnet, nhg_custom_primary))
+            {
+                if (!createNextHopGroup(vnet, nhg_custom_primary, vrf_obj, "custom"))
+                {
+                    SWSS_LOG_WARN("Failed to create primary based custom next hop group. Cannot proceed.");
+                    return;
+                }
+            }
+            updateRoute = true;
+        }
+        if (updateRoute)
+        {
+            nhg_custom = nhg_custom_primary;
+        }
+    }
+    else if (nhg_custom_secondary.getSize() > 0)
+    {
+        if (nhg_custom_secondary != active_nhg )
+        {
+            if (!hasNextHopGroup(vnet, nhg_custom_secondary))
+            {
+                if (!createNextHopGroup(vnet, nhg_custom_secondary, vrf_obj, "custom"))
+                {
+                    SWSS_LOG_WARN("Failed to create primary based custom next hop group. Cannot proceed.");
+                    return;
+                }
+            }
+            updateRoute = true;
+        }
+        if (updateRoute)
+        {
+            nhg_custom = nhg_custom_secondary;
+        }
+    }
+    else
+    {
+        //both HHG's are inactive, need to remove the route.
+        updateRoute = true;
+    }
+
+    if (nhg_custom.getSize() == 0)
+    {
+        // nhg_custom is empty. we shall create a dummy empty NHG for book keeping.
+        SWSS_LOG_INFO(" Neither Primary or Secondary endpoints are up.");
+        if (!hasNextHopGroup(vnet, nhg_custom))
+        {
+            NextHopGroupInfo next_hop_group_entry;
+            next_hop_group_entry.next_hop_group_id = SAI_NULL_OBJECT_ID;
+            next_hop_group_entry.ref_count = 0;
+            syncd_nexthop_groups_[vnet][nhg_custom] = next_hop_group_entry;
+        }
+    }
+    auto active_nhg_size = active_nhg.getSize();
+    if (updateRoute)
+    {
+        for (auto vr_id : vr_set)
+        {
+            if (nhg_custom.getSize() == 0)
+            {
+                if (active_nhg_size > 0)
+                {
+                    // we need to remove the route
+                    del_route(vr_id, pfx);
```

Comment:
Any reason to not check for del_route return value?

Reply:
What should we do if we check the status and the delete OP has failed?

Reply:
Perhaps add an ERR_LOG that can be identified later like it is done for update_route or add_route failure. Since updateVnetTunnelCustomMonitor returns a void, this error can't be propagated up.